src/FacultyDirectory/Helpers/AuthorizationHandler.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using FacultyDirectory.Core.Data;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.EntityFrameworkCore;
+
+namespace FacultyDirectory.Helpers
+{
+    public class AuthorizationHandler : AuthorizationHandler<AdminRequirement>
+    {
+        private readonly ApplicationDbContext dbContext;
+
+        public AuthorizationHandler(ApplicationDbContext dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+                                                       AdminRequirement requirement)
+        {
+            var username = context.User.Identity.Name;
+            var user = await dbContext.Users.Where(u => u.Username == username).FirstOrDefaultAsync();
```

Comment:
better to use `.Any()` here.  Users table is small so it doesn't make a big difference, but no reason to get the first row back when you just want a true/false.  Ex: `dbContext.Users.Any(u => u.Username == username)`